src/app/charts/charts.component.ts

```diff
@@ -89,15 +91,36 @@ export class ChartsComponent implements OnInit {
     });
   }
 
+  ngOnDestroy() {
+    // This ensures the IntervalObservable is cancelled
+    this.apiNotReady = false;
+  }
+
   loadCharts(): void {
-    this.chartsService.getCharts(this.repoName).subscribe(charts => {
+		this.chartsService.getCharts(this.repoName).subscribe(charts => {
       this.loading = false;
-      this.charts = charts;
-      if (!this.searchTerm) {
-        this.orderedCharts = this.orderCharts(this.charts);
-      }
+      this.setCharts(charts);
+    }, () => {
+      this.loading = false;
+      this.apiNotReady = true;
+      IntervalObservable.create(2000)
```

Comment:
The previous comments apply to this class too.

Reply:
The only thing needed here is the change to takeWhile right? The `setCharts` already de-duplicates the logic.